server/server.go

```diff
@@ -115,9 +116,17 @@ func New(cfg config.Config) (*Server, error) {
 
 	e := engine.New(reg, lManager, rStream, mach)
 
-	listeners, err := activation.Listeners(false)
-	if err != nil {
-		return nil, err
+	var listeners []net.Listener
+	if oldListeners == nil || len(oldListeners) == 0 {
+		listeners, err = activation.Listeners(false)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		listeners = make([]net.Listener, len(oldListeners))
+		for i, l := range oldListeners {
+			listeners[i] = l
+		}
```

Comment:
Why are you copying the values one by one: can't you just assign `listeners = oldListeners`?...

(This will just copy the reference rather than the actual values -- but I don't see why this would be undesirable?)